ui/app/AppLayouts/Communities/panels/SharedAddressesSigningPanel.qml

```diff
@@ -13,45 +13,66 @@ import SortFilterProxyModel 0.2
 ColumnLayout {
     id: root
 
+    required property string componentUid
+    required property bool isEditMode
     property var keypairSigningModel
 
-    readonly property string title: qsTr("Prove ownership of keypairs")
+    required property var selectedSharedAddressesMap // Map[address, [keyUid, selected, isAirdrop]
+    required property int totalNumOfAddressesForSharing
+
+    required property string communityName
+    readonly property string title: root.isEditMode?
+                                        qsTr("Save addresses you share with %1").arg(root.communityName)
+                                      : qsTr("Request to join %1").arg(root.communityName)
     readonly property var rightButtons: [d.rightBtn]
-    readonly property bool allSigned: regularKeypairs.visible == d.sharedAddressesForAllNonKeycardKeypairsSigned &&
-                                      keycardKeypairs.visible == d.allKeycardKeypairsSigned
 
     signal joinCommunity()
-    signal signSharedAddressesForAllNonKeycardKeypairs()
+    signal signProfileKeypairAndAllNonKeycardKeypairs()
     signal signSharedAddressesForKeypair(string keyUid)
 
-    function sharedAddressesForAllNonKeycardKeypairsSigned() {
-        d.sharedAddressesForAllNonKeycardKeypairsSigned = true
+    function allSigned() {
+        d.allSigned = true
     }
 
     QtObject {
         id: d
 
-        property bool sharedAddressesForAllNonKeycardKeypairsSigned: false
-        property bool allKeycardKeypairsSigned: false
+        readonly property int selectedSharedAddressesCount: root.selectedSharedAddressesMap.size
 
-        readonly property var rightBtn: StatusButton {
-            enabled: root.allSigned
-            text: qsTr("Share your addresses to join")
-            onClicked: {
-                root.joinCommunity()
+        property bool allSigned: false
+
+        readonly property bool anyOfSelectedAddressesToRevealBelongToProfileKeypair: {
+            for (let [key, value] of root.selectedSharedAddressesMap) {
+                if (value.keyUid === userProfile.keyUid) {
+                    return true
+                }
             }
+            return false
         }
 
-        function reEvaluateSignedKeypairs() {
-            let allKeypairsSigned = true
-            for(var i = 0; i< keycardKeypairs.model.count; i++) {
-                if(!keycardKeypairs.model.get(i).keyPair.ownershipVerified) {
-                    allKeypairsSigned = false
-                    break
+        readonly property bool thereAreMoreThanOneNonProfileRegularKeypairs: nonProfileRegularKeypairs.count > 1
+
+        readonly property bool allNonProfileRegularKeypairsSigned: {
```

Comment:
Does it mean, the rest of keypairs NOT dependent on the profile keypair authentication (meaning that, all possible keycard ones) are already signed?

Reply:
Not exactly, it refers that all keypair that are not profile keypairs and that are not migrated to a Keycard are signed.